src/modules/formBuilder/components/FormEditorItemPreview.tsx

```diff
@@ -1,24 +1,36 @@
 import { Card, Typography } from '@mui/material';
+import { useEffect, useState } from 'react';
+import { Control, DeepPartial, useWatch } from 'react-hook-form';
 import theme from '@/config/theme';
 import DynamicField from '@/modules/form/components/DynamicField';
 import DynamicViewField from '@/modules/form/components/viewable/DynamicViewField';
 import { FormItem } from '@/types/gqlTypes';
 
-interface FormEditorItemPreviewProps {
-  item: FormItem;
-  value?: any;
-  setValue: (value: any) => void;
+interface Props {
+  control: Control<DeepPartial<FormItem>>;
 }
+export const FormEditorItemPreview: React.FC<Props> = ({ control }) => {
+  // watch whole form. probably doesn't need to watch everything (like enablewhen, etc). can pass `name` to specify fields to watch
```

Comment:
I think watching the whole form seems fine. If we do decide to pass `name` and restrict the list of fields it watches, then maybe it would be good to add a comment here explaining why (my understanding is that we only need to watch the fields that could impact this individual form item's display, since the preview only shows one item so it can't be used for previewing conditional logic like enableWhen)

Reply:
Yeah, agree that watching the whole item is okay. If you pass `name` then it returns an array of the values in the same order, so that will just get super wordy since there are so many attributes. I'll leave as is and update the comment!